tests/UI/dashboardUI.spec.js

```diff
@@ -1,30 +1,70 @@
 import { expect } from "@playwright/test";
 import { test } from "../../fixtures/basePage.js";
+import { CustomersUI } from "../../modules/customersUI.js";
 import { AuthUI } from "../../modules/authUI.js";
 import { URLS } from "../../fixtures/urls.js";
 
 test.describe("Dasboard tests", () => {
   let page;
   let authUI;
+  let customersUI;
 
   test.beforeAll("Log in", async ({ browser }) => {
     page = await browser.newPage();
     authUI = new AuthUI(page);
+    customersUI = new CustomersUI(page);
     await page.goto(URLS["LOGIN_PAGE"]);
     await authUI.login({ valid: true });
   });
 
+  test.beforeEach(
+    "Visit the dashboard page and wait for all products to load",
+    async ({}) => {
+      // We wait for all products to load because the tests run
+      // too fast otherwise and innacurate results are recieved
+
+      await page.goto(URLS["DASHBOARD"]);
+      await expect(page).toHaveURL(URLS["DASHBOARD"]);
+      await expect(customersUI.spinner).toBeVisible();
+      await customersUI.spinner.waitFor({ state: "hidden" });
+      for (const prod of await customersUI.product.all()) {
+        await expect(prod).toBeVisible();
+      }
+    }
+  );
+
   test.afterAll("Close page", async ({}) => {
     await page.close();
   });
 
   test("Add a product to cart", { tag: "@smoke" }, async ({}) => {
-    await page.goto(URLS["DASHBOARD"]);
-    await expect(page).toHaveURL(URLS["DASHBOARD"]);
+    await customersUI.addProductToCart({ productID: 20 });
```

Comment:
ok su testovi kao testovi, ali nemoj imati dashboard tests, fokusiraj se na funkcionalnosti... tipa bilo bi dobro da imamo
- filters tests
- pagination tests
- search tests
- add to cart tests
- buy products tests

etc... znaci vezemo se za funkcionalnosti... Tako da bi bilo idealno da ovo razdvojis

Reply:
takodje onda u svakoj funkcionalnosti imas jedan test koji je smoke, ostatak pripada regresiji itd